Utilities/macrecovery/macrecovery.py

```diff
@@ -208,9 +272,10 @@ def action_download(args):
     print(info)
   print('Downloading ' + info[INFO_PRODUCT] + '...')
   dmgname = '' if args.basename == '' else args.basename + '.dmg'
-  save_image(info[INFO_IMAGE_LINK], info[INFO_IMAGE_SESS], dmgname, args.outdir)
+  dmgname = save_image(info[INFO_IMAGE_LINK], info[INFO_IMAGE_SESS], dmgname, args.outdir)
   cnkname = '' if args.basename == '' else args.basename + '.chunklist'
-  save_image(info[INFO_SIGN_LINK], info[INFO_SIGN_SESS], cnkname, args.outdir)
+  cnkname = save_image(info[INFO_SIGN_LINK], info[INFO_SIGN_SESS], cnkname, args.outdir)
+  verify_image(dmgname, cnkname)
```

Comment:
Could you please add exception handling here? Would not be great to have signature verification failure look like a crash.

Reply:
Since none of the asserts have any exception text, I thought it might be worth displaying the assertion that failed.

<details>
<summary>temp code to corrupt chunklist (inserted in action_download at line ~279-ish)</summary>

```
  path = cnkpath
  count = int(random.random() * 10)
  offset = 10 # int((os.stat(path).st_size - count) * random.random())
  with open (path, 'r+b') as f:
    f.seek(offset)
    f.write(random.randbytes(count))
  print(f'Wrote {count} bytes at offset {offset} to {path}')
```
</details>

Sample output:
```
Downloading 694-35959...
Saving http://oscdn.apple.com/content/downloads/41/28/071-71279/5djpevqmc2w0ti2c4n1m8mkz7fm791t1ic/RecoveryImage/BaseSystem.dmg to BaseSystem.dmg...
Download complete!MBs downloaded...
Saving http://oscdn.apple.com/content/downloads/41/28/071-71279/5djpevqmc2w0ti2c4n1m8mkz7fm791t1ic/RecoveryImage/BaseSystem.chunklist to BaseSystem.chunklist...
Download complete!25 MBs downloaded...
Wrote 4 bytes at offset 10 to /Users/user/Downloads/dev/OpenCorePkg/Utilities/macrecovery/BaseSystem.chunklist
Verifying image with chunklist...
Image verification failed. (assert signature_method in [1, 2])
```

Reply:
Looks nice, thank you!